rbtree_bench_test.go

```diff
@@ -0,0 +1,64 @@
+package rbtree
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+)
+
+func BenchmarkRBTree(b *testing.B) {
+	tree := NewTree[int64, string]()
```

Comment:
There is only one instance of a `tree` in the entire benchmark, causing each `insert` subbenchmark  to add more key-value pairs. As a result, the `tree` size isn't always same as `b.N`. 
And the length in `iterate` subbenchmark  is always a fixed value.

I suggest creating a new tree in each subbenchmark, and inserting data before `b.ResetTimer` in the iterate and delete subbenchmarks.

Reply:
You're right. Fixed. Apologies for all the back and forth on this.

Reply:
No worries. Thank you for your pull request.